question/bank/managecategories/category_class.php

```diff
@@ -0,0 +1,609 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * A class for representing question categories.
+ *
+ * @package    moodlecore
+ * @subpackage questionbank
+ * @copyright  1999 onwards Martin Dougiamas {@link http://moodle.com}
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+
+defined('MOODLE_INTERNAL') || die();
+
+// number of categories to display on page
+define('QUESTION_PAGE_LENGTH', 25);
+
+require_once($CFG->libdir . '/listlib.php');
+require_once($CFG->dirroot . '/question/bank/managecategories/category_form.php');
+require_once($CFG->dirroot . '/question/bank/managecategories/move_form.php');
+
+
+/**
+ * Class representing a list of question categories
+ *
+ * @copyright  1999 onwards Martin Dougiamas {@link http://moodle.com}
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class question_category_list extends moodle_list {
```

Comment:
create a different file which will contain only this class and point the code to that, also namespace it against the plugin.